sys/include/net/ng_pktbuf.h

```diff
@@ -59,68 +59,58 @@ extern "C" {
- *          of ng_pktsnip_t of being reversed for either the sending or
- *          receiving context. Because of this the assumtion of the transmission
- *          direction, the state of the packet buffer and the values for the
- *          members of the resulting ng_pktsnip_t can be very different after
- *          execution of this function depending on what parameters you use:
- *
- * * The return value of this function is a @ref ng_pktsnip_t struct referred
- *   to as `result`
- * * for most cases the build-up of `result` will be pretty straight forward: Its
- *   members will be exactly as the given parameters (ng_pktsnip_t::next of
- *   result will be set to @p pkt). If @p pkt is not NULL it and in turn `result`
- *   are assumed to be in sending direction. For packet creation (@p pkt == NULL)
- *   no assumtions about direction of `result` will be made (since its
- *   ng_pktsnip::next will be set to NULL).
- * * if @p pkt != NULL, @p data = `pkt->data`, @p size < `pkt->size` receiving
- *   direction is assumed and the following values will be set:
- *   * ng_pktsnip_t::next of `result` = `pkt->next`
- *   * ng_pktsnip_t::data of `result` = @p data
- *   * ng_pktsnip_t::size of `result` = @p size
- *   * ng_pktsnip_t::next of @p pkt = `result`
- *   * ng_pktsnip_t::data of @p pkt = @p data + @p size
- *   * ng_pktsnip_t::size of @p pkt = old size value - @p size
- *
- *  graphically this can be represented as follows:
- *  @code
- * Before                                    After
- * ======                                    =====
- *                                                       (next)
- *  pkt->data                                 result->data <== pkt->data
- *  v                                         v                v
- * +--------------------------------+        +----------------+---------------+
- * +--------------------------------+        +----------------+---------------+
- *  \__________pkt->size___________/          \_result->size_/ \__pkt->size__/
- *  @endcode
- *
- * @note    **Do not** change the ng_pktsnip_t::data and ng_pktsnip_t::size
+ * @warning **Do not** change the ng_pktsnip_t::data and ng_pktsnip_t::size
  *          of a ng_pktsnip_t created by this function externally, except if
  *          they both are null or data is not from inside the packet buffer.
  *          This will most likely create memory leaks.
  *
- * @param[in,out] pkt   The packet you want to add a ng_pktsnip_t to. Leave
- *                      NULL if you want to create a new packet. Members may
- *                      change values; see above.
+ * @param[in] next      Next ng_pktsnip_t in the packet. Leave NULL if you
+ *                      want to create a new packet.
  * @param[in] data      Data of the new ng_pktsnip_t. If @p data is NULL no data
  *                      will be inserted into `result`. If @p data is already
  *                      in the packet buffer (e.g. a payload of an already
```

Comment:
I would vote for also simplifying this and copy data in any case, so we drop the checks if data is already in the buf...